### PR TITLE
Add nvprims `rand_like` support for Dropout

### PR DESCRIPTION
**Pull request by kevinstephano**
NM

### CODE REVIEW

File: torch/_prims/context.py
Comment (IvanYashchuk) [inline torch/_prims/context.py:238]: I should change it in my view PR. Let's state explicitly things we catch instead of substring matching
```suggestion
            func == torch.ops.aten.rand_like or func == torch.ops.aten.rand_like.default
```

**Review by IvanYashchuk — COMMENTED** [review]
Adding `PythonRefInfo("ops.nvprims.rand_like", ...)` to common_methods_invocations.py file would enable tests that can be run with `python -m pytest test/test_ops.py -k "nvprims_rand_like"`, but due to randomness they're most likely to fail. Maybe for random functions we should disable comparison of values and check only shape, dtype, device?

File: test/test_prims.py
Comment (IvanYashchuk) [inline test/test_prims.py:194]: nit: strictly_nvfuser is not a partitioned executor 😄 Everything that's in `gm` is sent to nvFuser's execution in this case.

File: torch/_prims/context.py
Comment (IvanYashchuk) [inline torch/_prims/context.py:253]: So all keyword arguments (dtype, layout, device, pin_memory, memory_format) are ignored. Should we give at least a warning in this case (`len(kwargs) > 0`)?

**Review by IvanYashchuk — APPROVED** [review]
Thanks for adding this! I'm only worried about silently ignoring keyword arguments.

**Comment by IvanYashchuk** [comment]
@mruberry could we please get your approval? Note that having `rand_like` as a primitive for nvFuser is most likely a temporary solution needed for dropout. We have support for the `uniform` primitive, that would be used for `refs.rand_like`, in our devel branch, it just hasn't made its way to this repo yet.

File: torch/_prims/nvfuser_prims.py
Comment (ngimel) [inline torch/_prims/nvfuser_prims.py:332]: these are not correct strides for rand_like?

File: torch/_prims/context.py
Comment (kevinstephano) [reply torch/_prims/context.py:256]: Added a warning.

File: test/test_prims.py
Comment (kevinstephano) [reply test/test_prims.py]: Removed the comment.

File: torch/_prims/nvfuser_prims.py
Comment (kevinstephano) [reply torch/_prims/nvfuser_prims.py:332]: We currently don't support maintaining strided outputs in the nvFuser integration.  Given that restriction, is this still incorrect to do?

Comment (jjsjann123) [reply torch/_prims/nvfuser_prims.py:332]: We never get the bandwidth to plumb memory format support through. For now, codegen only produces contiguous tensor. Luckily nvfuser_prims are only used by nvfuser, so we are not breaking others.

There were conversations about respecting memory format and that is on our radar. For now I think we can just leave a TODO here.

**Review by mruberry — APPROVED** [review]
Approving to unblock.

We'll have these prims in primTorch very soon, and we should probably start more regularly and detailed integration discussions now that 1.13 is winding down.

**Pull request closed without merging**
